framework/aster-frame/src/sync/rcu/mod.rs

```diff
@@ -32,13 +90,38 @@ impl<P: OwnerPtr> Rcu<P> {
         }
     }
 
+    /// Retrieves a read guard for the RCU mechanism.
+    ///
+    /// This method returns a `RcuReadGuard` which allows read-only access to the
+    /// underlying data protected by the RCU mechanism.   
+    ///
+    /// This function has the semantics of _subscribe_ in RCU mechanism.
+    ///
+    /// # Safety
+    ///
+    /// The pointer protected by the Rcu must be valid and point to a valid object.
+    ///
+    /// # Non-Preemptible RCU
+    ///
+    /// In non-preemptible RCU, the read-side critical section is delimited using
+    /// a `PreemptGuard`.
+    // TODO: Distinguish different type RCU
     pub fn get(&self) -> RcuReadGuard<'_, P> {
+        let guard = disable_preempt();
         let obj = unsafe { &*self.ptr.load(Acquire) };
-        RcuReadGuard { obj, rcu: self }
+        RcuReadGuard {
+            obj,
+            rcu: self,
+            inner_guard: InnerGuard::Preempt(guard),
```

Comment:
`preempt_guard` makes a better name than `inner_guard` as one can tell the purpose of the variable from the name.